var/spack/repos/builtin/packages/umpire/package.py

```diff
@@ -458,37 +456,56 @@ def initconfig_package_entries(self):
     def cmake_args(self):
         return []
 
-    def test(self):
+    def run_umpire(self, exe, expected):
         """Perform stand-alone checks on the installed package."""
-        if self.spec.satisfies("@:1") or not os.path.isdir(self.prefix.bin):
-            tty.info("Skipping: checks not installed in bin for v{0}".format(self.version))
-            return
-
-        # Run a subset of examples PROVIDED installed
-        # tutorials with readily checkable outputs.
-        checks = {
-            "malloc": ["99 should be 99"],
-            "recipe_dynamic_pool_heuristic": ["in the pool", "releas"],
-            "recipe_no_introspection": ["has allocated", "used"],
-            "strategy_example": ["Available allocators", "HOST"],
-            "tut_copy": ["Copied source data"],
-            "tut_introspection": ["Allocator used is HOST", "size of the allocation"],
-            "tut_memset": ["Set data from HOST"],
-            "tut_move": ["Moved source data", "HOST"],
-            "tut_reallocate": ["Reallocated data"],
-            "vector_allocator": [""],
-        }
-
-        for exe in checks:
-            expected = checks[exe]
-            reason = "test: checking output from {0}".format(exe)
-            self.run_test(
-                exe,
-                [],
-                expected,
-                0,
-                installed=False,
-                purpose=reason,
-                skip_missing=True,
-                work_dir=self.prefix.bin,
-            )
+        if self.spec.satisfies("@:1"):
+            raise SkipTest("Package must be installed as version @1.0.1 or later")
+
+        if os.path.isdir(self.prefix.bin):
+            raise SkipTest(f"{self.prefix.bin} does not eixst")
+
+        exe_run = which(join_path(self.prefix.bin, exe))
+        if exe_run is None:
+            raise SkipTest("Executable not present within directory")
```

Comment:
The message at line 469 doesn't say what directory is missing the executable.

Just check that the path to the installed executable exists.  There's no reason to check versions, the presence of the `bin` subdirectory, or if an `Executable` instance is created (all in separate checks).

Note: Building with `tests=basic` for at least the latest version makes the `bin` subdirectory but that executable isn't being tested here.  So my suggestion covers this base as well.

Reply:
It looks like newer versions -- since version `4` -- require `examples` to be enabled to build the programs checked here.  I'm not sure if it's worth noting that though.

@davidbeckingsale @adrienbernede What are your thoughts on whether stand-alone testing should be informing people they have to build with certain options (or versions+options) to get stand-alone tests?